### ISSUE TITLE
Infinite scroll is unable to handle items of different sizes - fix

### ISSUE THREAD

**Issue by SimonDarksideJ** [Unity-UI-Extensions/com.unity.uiextensions#466]
<sup>Issue created by @SimonDarksideJ as [Bitbucket Issue #&#8203;327](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/327) on 2020.07.13 11:04.</sup>
**Raised on behalf of @Jairo Andrés Pinzon**

Hola, no hablo ingles espero que esta traducción de google pueda ser de ayuda!

descargue el script pero no me funcionaba con elementos de distintos tamaños, probe esto y se corrigió \(solo modifique el movimiento horizontal\)!

---

Hi, I do not speak English I hope this google translation can be helpful!

I downloaded the script but it didn't work with elements of different sizes, I tried this and it was corrected \(just modify the horizontal movement\)!

---

```
public void OnScroll(Vector2 pos)
{
    if (!_hasDisabledGridComponents)
        DisableGridComponents();

    for (int i = 0; i < items.Count; i++)
    {
        if (_isHorizontal)
        {
            if (_scrollRect.transform.InverseTransformPoint(items[i].gameObject.transform.position).x > _disableMarginX + 100)
            {
                _newAnchoredPosition = items[i].anchoredPosition;
                _newAnchoredPosition.x = items[0].anchoredPosition.x - items[i].sizeDelta.x - 20;

                items[i].anchoredPosition = _newAnchoredPosition;
                _scrollRect.content.GetChild(_itemCount - 1).transform.SetAsFirstSibling();

                SetItems();                
            }
            else if (_scrollRect.transform.InverseTransformPoint(items[i].gameObject.transform.position).x < -_disableMarginX -100)
            {
                _newAnchoredPosition = items[i].anchoredPosition;
                _newAnchoredPosition.x = items[_itemCount - 1].anchoredPosition.x + items[_itemCount - 1].sizeDelta.x + 20;

                items[i].anchoredPosition = _newAnchoredPosition;
                _scrollRect.content.GetChild(0).transform.SetAsLastSibling();

                items.Clear();

                for (int i = 0; i < _scrollRect.content.childCount; i++)
                {
                    items.Add(_scrollRect.content.GetChild(i).GetComponent<RectTransform>());
                }
            }
        }

        //if (_isVertical)
        //{
        //    if (_scrollRect.transform.InverseTransformPoint(items[i].gameObject.transform.position).y > _disableMarginY + 200)
        //    {
        //        _newAnchoredPosition = items[i].anchoredPosition;
        //        _newAnchoredPosition.y = items[0].anchoredPosition.y - items[i].sizeDelta.y - 20;

        //        items[i].anchoredPosition = _newAnchoredPosition;
        //        _scrollRect.content.GetChild(_itemCount - 1).transform.SetAsFirstSibling();

        //        SetItems();
        //    }
        //    else if (_scrollRect.transform.InverseTransformPoint(items[i].gameObject.transform.position).y < -_disableMarginY -200)
        //    {
        //        _newAnchoredPosition = items[i].anchoredPosition;
        //        _newAnchoredPosition.y = items[_itemCount - 1].anchoredPosition.y + items[_itemCount - 1].sizeDelta.y + 20;

        //        items[i].anchoredPosition = _newAnchoredPosition;
        //        _scrollRect.content.GetChild(0).transform.SetAsLastSibling();

        //        SetItems();

        //        Debug.Log("abajo");
        //    }
        //}
    }
}
```

‌

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.07.13 11:17](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/327#comment-58145709), @SimonDarksideJ commented:</sup>
@{5a943f3829334854a7c6a96d} ¿Puede proporcionar el script completo, ya que sus ejemplos se refieren a una función “SetItems” que no existe?

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.07.13 14:00](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/327#comment-58147402), Jairo Andrés Pinzon commented:</sup>
Hola, esto hace la función, si necesitas algo mas me avisas.

Hello, this does the function, if you need something else let me know.

```
private void SetItems()
{
    items.Clear();
    for (int i = 0; i < _scrollRect.content.childCount; i++)
    {
        items.Add(_scrollRect.content.GetChild(i).GetComponent<RectTransform>());
    }
}
```

‌

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.07.13 20:31](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/327#comment-58151357), @SimonDarksideJ commented:</sup>
@{5a943f3829334854a7c6a96d} 

OK, working through the issue, but I do not think your solution works. It does not take in to account the different child sizes to determine when an item should move and it is fixing the anchors instead of altering them based on the movement.

I will work out some math and get back to you. Interesting problem.

---

OK, trabajando en el problema, pero no creo que su solución funcione. No tiene en cuenta los diferentes tamaños de niños para determinar cuándo se debe mover un elemento y está fijando los anclajes en lugar de alterarlos según el movimiento.

Resolveré algunas matemáticas y te responderé. Interesante problema

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.10.14 17:03](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/327#comment-59048346), @SimonDarksideJ commented:</sup>
@{5cb624753704fd1181fed49b} has provided an updated implementation for the above with seems to work better in #349 , so closing this with those changes.

‌

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.10.14 17:04](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/327#comment-59048351), @SimonDarksideJ commented:</sup>
Resolved by #349

**Comment by SimonDarksideJ** [comment]
<sup>On 2020.10.14 17:04 @SimonDarksideJ modified issue:</sup>
**status** changed `new` → `resolved`
